esmvaltool/_task.py

```diff
@@ -462,15 +467,29 @@ def _run(self, input_files):
 
         process = self._start_diagnostic_script(cmd, env)
 
+        returncode = None
+        last_line = ['']
+
         with resource_usage_logger(process.pid, self.resource_log),\
                 open(self.log, 'at') as log:
-            for line in process.stdout:
-                log.write(line)
+            while returncode is None:
+                returncode = process.poll()
+                txt = process.stdout.read()
+                txt = txt.decode(encoding='utf-8', errors='ignore')
```

Comment:
no need to set the encoding, it's picked up automatically from `locale` and if the locale encoding is not UTF-8 it's not safe to set it at decode level